modules/objc/VectorSeachSnippets.m

```diff
@@ -0,0 +1,120 @@
+//
+//  VectorSeachSnippets.m
+//
+//  Created by Vlad Velicu on 20/03/2024.
+//
+
+#import <Foundation/Foundation.h>
+#import <CouchbaseLite/CouchbaseLite.h>
+#import <NaturalLanguage/NLEmbedding.h>
+
+CBLDatabase* database;
+CBLCollection* collection;
+NLEmbedding* model;
+
+// MARK: Configuring a project to use Vector Search.
+
+/*/
+ Download Couchbase Lite iOS Obj-C and Couchbase Lite Vector Search libraries
+ Extract the xcframeworks and copy them into the App Project Directory
+ 
+ Add both libraries to the *Frameworks, Libraries and Embedded Content* to your desired target
+ */
+
+@interface VectorSearch : NSObject 
+
+@end
+
+@implementation VectorSearch
+
+- (void) createDefaultIndex {
+    NSError* error;
+    database = [[CBLDatabase alloc] initWithName:@"vectorDB" error:&error];
+    collection = [database defaultCollection: &error];
+    
+    // MARK: Create a default Vector Index Configuration
+
+    CBLVectorIndexConfiguration* config = [[CBLVectorIndexConfiguration alloc] initWithExpression: @"vector" dimensions: 300 centroids: 20];
+    
+    // Default values already set:
+    NSLog(@"%@", config.encoding); // .scalarQuantizer(type: .SQ8)
```

Comment:
I'm not sure if it will print out a useful info.

Reply:
swift is fine.
obj-c is giving the type and the mem address... `CBLScalarQuantizer:<...>`. I couldn't find a more suitable way to show the default values - I am open for suggestions

Reply:
If we explain this in the doc content for this code snippet, we don't need to have all ofthe NSLog to show the default value at all.

Reply:
I think it's better to describe it in the content instead of showing it as a comment in the code. So let remove it and add it back if we really need it.